### PR TITLE
docs: Add debug pause docs. Fixes #7684

### LINKED ISSUE #7684: Debug pause docs
* Do we have docs for this feature? I see e2e tests but no docs or example in the example folder.

### PR DESCRIPTION
**Pull request by NikeNano**
Signed-off-by: NikeNano <niklas.sven.hansson@gmail.com>

Fixes #7684

### CODE REVIEW

File: docs/debug-pause.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Debug pause
 
-Argo v3.3 introduced the possibility to pause steps for debugging. This feature is currently only supported when using the emissary workflow executor[Emissary Executor](workflow-executors.md#emissary-emissary)
+> v3.3  and after
 
-The debug pause feature makes it possible to pause individual steps for debugging before, after or both and then release the steps from the paused state. 
+## Introduction
+
+This feature makes it easier to debug workflow steps by allowing users to specify if a steps should be paused. Currently this feature is only supported when using the emissary workflow executor[Emissary Executor](workflow-executors.md#emissary-emissary)
```
Comment (terrytangyuan) [inline docs/debug-pause.md:7]: Redundant “emissary workflow executor”

File: docs/debug-pause.md [hunk 2]
```diff
@@ -61,9 +63,9 @@ argo submit -n argo --watch pause-after.yaml
 kubectl debug -n argo -it POD_NAME --image=busybox --target=main --share-processes
 ```
 
-In order to have access to the persistance volume used but the workflow step,  [`--share-processes`](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/) will have to be used. 
+In order to have access to the persistance volume used by the workflow step,  [`--share-processes`](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/) will have to be used. 
 
-When inside the shell of the ephemeral container deugging of the pod can be done, when done the marker file can be create in order to allow the step continue to run.
+When inside the shell of the ephemeral container deugging of the pod can be done, when done the marker file can be create in order to allow the step continue to run. To create the marker file to release the step:
```
Comment (terrytangyuan) [inline docs/debug-pause.md:68]: Typo: debugging

File: docs/debug-pause.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Debug pause
+
+> v3.3  and after
```
Comment (blkperl) [inline docs/debug-pause.md:3]:
```suggestion
> v3.3 and after
```

File: docs/debug-pause.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Introduction
+
+This feature makes it easier to debug workflow steps by allowing users to specify if a steps should be paused. Currently this feature is only supported when using the emissary workflow executor[Emissary Executor](workflow-executors.md#emissary-emissary)
```
Comment (blkperl) [inline docs/debug-pause.md:7]:
```suggestion
This feature makes it easier to debug workflow steps by allowing users to specify if a steps should be paused. Currently this feature is only supported when using the [Emissary Executor](workflow-executors.md#emissary-emissary)
```

File: docs/debug-pause.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  templates:
+    - name: whalesay
+      container:
+        image: argoproj/argosay:v2
+        env:
+          - name: ARGO_DEBUG_PAUSE_AFTER
+            value: 'true'
+```
+
+In order to release a step from a pause state,  marker files are used named `after` or `before` corresponding to when the step is paused. Pausing steps can with benefits be used together with [ephemeral containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) when a shell is not available in the used container.
```
Comment (blkperl) [inline docs/debug-pause.md:33]:
```suggestion
In order to release a step from a pause state,  marker files are used named `after` or `before` corresponding to when the step is paused. Pausing steps can be used together with [ephemeral containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) when a shell is not available in the used container. 
```

nit: `with benefits` seems redundant

File: docs/debug-pause.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
+# Debug pause
+
+> v3.3  and after
+
+## Introduction
+
+This feature makes it easier to debug workflow steps by allowing users to specify if a steps should be paused. Currently this feature is only supported when using the emissary workflow executor[Emissary Executor](workflow-executors.md#emissary-emissary)
+
+The `debug pause` feature makes it possible to pause individual workflow steps for debugging before, after or both and then release the steps from the paused state.
```
Comment (blkperl) [inline docs/debug-pause.md:9]: This restates line 7 can we merge them together?

File: docs/debug-pause.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+2) Create a shell in the container of interest of create a ephemeral container in the pod, in this example ephemeral containers are used. 
+
+```
+kubectl debug -n argo -it POD_NAME --image=busybox --target=main --share-processes
+```
+
+In order to have access to the persistance volume used by the workflow step,  [`--share-processes`](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/) will have to be used. 
+
+When inside the shell of the ephemeral container deugging of the pod can be done, when done the marker file can be create in order to allow the step continue to run. To create the marker file to release the step:
```
Comment (blkperl) [inline docs/debug-pause.md:68]: I think this sentence is hard to parse. We could reword this to:

The ephemeral container can be used to perform debugging operations. When debugging has been completed, create the marker file to allow the workflow step to continue:

**Comment by alexec** [comment]
You should also add docs/debug-pause.md to mkdocs.yaml

File: docs/debug-pause.md (hunk 7 above)
Comment (NikeNano) [reply docs/debug-pause.md:68]: Good suggestion, will update

File: mkdocs.yml [hunk 8]
```diff
@@ -88,6 +88,7 @@ nav:
           - resuming-workflow-via-automation.md
           - async-pattern.md
           - running-at-massive-scale.md
+          - debug-paused.md
```
Comment (alexec) [inline mkdocs.yml:91]: typo

**Review by terrytangyuan — APPROVED** [review]

File: docs/debug-pause.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  templates:
+    - name: whalesay
+      container:
+        image: argoproj/argosay:v2
+        env:
+          - name: ARGO_DEBUG_PAUSE_AFTER
+            value: 'true'
+```
+
+In order to release a step from a pause state,  marker files are used named `after` or `before` corresponding to when the step is paused. Pausing steps can be used together with [ephemeral containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) when a shell is not available in the used container.
```
Comment (alexec) [inline docs/debug-pause.md:31]: full path of marker files please

**Review by alexec — APPROVED** [review]
